src/Decompiler/Analysis/DataFlowAnalysis.cs

```diff
@@ -87,17 +87,22 @@ public void BuildExpressionTrees()
                     var sst = new SsaTransform(flow, proc, importResolver, doms, new HashSet<RegisterStorage>());
                     var ssa = sst.SsaState;
 
-                    var icrw = new IndirectCallRewriter(program, ssa, eventListener);
-                    icrw.Rewrite();
+                    var vp = new ValuePropagator(program.Architecture, ssa);
 
+                    var icrw = new IndirectCallRewriter(program, ssa, eventListener);
+                    while (!eventListener.IsCanceled() && icrw.Rewrite())
+                    {
+                        vp.Transform();
+                        sst.RenameFrameAccesses = true;
```

Comment:
Consider moving this out if the loop; it's not necessary to reset the property once it is set.